### PR TITLE
DOC: Add dos and donts for writing change handlers

### PR DESCRIPTION
**Pull request by kitchoi**
This PR adds a Dos and Donts' subsection in the `Trait Notification` section and add one example.

**Checklist**
- ~Tests~
- ~Update API reference (`docs/source/traits_api_reference`)~
- [x] Update User manual (`docs/source/traits_user_manual`)
- ~Update type annotation hints in `traits-stubs`~

### CODE REVIEW

File: docs/source/traits_user_manual/notification.rst [hunk 1]
```diff
@@ -719,3 +719,47 @@ steps other than 1, **index** holds the _slice_ that was changed.
...
+
+The first example is problematic because when ``name`` changes, calling
+``update_orders`` after ``update_number``  produces a result that is different
+from calling ``update_number`` after ``update_orders``.
+
+Even if the change handlers appear to be called in a deterministic order,
+this would be due to implementation details that may not hold true across
+releases and platforms.
+
+Traits consider handlers for the same change event to be independent of each
```
Comment (mdickinson) [inline docs/source/traits_user_manual/notification.rst:760]: Do you think we can break this paragraph (or part of it) out into a separate "don't", namely: "don't raise exceptions from change handlers"? Or maybe that's better done in a separate PR?

Comment (mdickinson) [reply docs/source/traits_user_manual/notification.rst:760]: Also, it's kinda obvious, but "don't return values from change handlers" might be worth having, too.

Comment (kitchoi) [reply docs/source/traits_user_manual/notification.rst:760]: I imagine this section to carry on growing 🌴.  Yes I can separate these two out here.

Comment (kitchoi) [reply docs/source/traits_user_manual/notification.rst:778]: Actually, I ended up not adding "don't return values from change handlers".  It is unlikely anyone has been surprised by it so it might end up clustering the section.

File: docs/source/traits_user_manual/notification.rst [hunk 2]
```diff
@@ -719,3 +719,62 @@ steps other than 1, **index** holds the _slice_ that was changed.
...
 values can be queried from directly from the trait value, if needed).
+
+
+.. _on-trait-change-dos-n-donts:
+
+
+Dos and Don’ts
+--------------
+
+Do not assume handlers are called in a specific order
```
Comment (kitchoi) [inline docs/source/traits_user_manual/notification.rst:730]: Nit: I am going to use "Don't" instead of "Do not", just to be consistent with the title.
